packages/ckeditor5-table/src/converters/table-cell-paragraph-post-fixer.ts

```diff
@@ -100,7 +100,7 @@ function fixTableRow( tableRow: Element, writer: Writer ) {
  */
 function fixTableCellContent( tableCell: Element, writer: Writer ) {
 	// Insert paragraph to an empty table cell.
-	if ( tableCell.childCount == 0 ) {
+	if ( tableCell.childCount == 0 && !tableCell.hasAttribute( 'htmlEmptyBlock' ) ) {
```

Comment:
The model structure should always be normalized. The paragraphs should be there. Note that table cells and list items use "bogus" paragraphs, those are not rendered in the data pipeline and are rendered as a `<span>` in the editing pipeline. Maybe the solution should alter bogus paragraph handling, not the model structure.